xds/src/main/java/io/grpc/xds/XdsResourceType.java

```diff
@@ -85,9 +85,7 @@
 
   abstract String typeUrlV2();
 
-  // Non-null for  State of the World resources.
-  @Nullable
-  abstract XdsResourceType<?> dependentResource();
+  abstract boolean isStateOfTheWorld();
```

Comment:
As silly as it seems, Envoy uses the "state of the world" terminology for all resource types when using the non-delta protocol (which is what we are doing). SotW is an aspect of the RPC method and not the resource type. It's just that some resource types behave differently in SotW from each other...

https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol

> The SotW approach was the original mechanism used by xDS, in which the client must specify all resource names it is interested in with each request, and for LDS and CDS resources, the server must return all resources that the client has subscribed to in each request.

So we should use a different word for this method. I don't know what to name it though. "isFullStateOfTheWorld"? Throw a short javadoc on the method to resolve any source of ambiguity.